### ISSUE TITLE
Handle thumbnail creation of video files correctly

### ISSUE THREAD

**Issue by tomamplius** [LibrePhotos/librephotos#211]
On the commit https://github.com/LibrePhotos/librephotos/commit/187cfa2da9c04330cbe3402ad4b2ebd3408aff23
is_valid_media is remove then all file are processed.

On my folder there is mp4 video are processed with the bellow error : 
~~~
20200118_195052.mp4

reason: delegate failed `'ffmpeg' -nostdin -v -1 -i '%i' -vframes %S -vcodec pam -an -f rawvideo -y '%u.pam' 2> '%u'' @ error/delegate.c/InvokeDelegate/1966
Traceback (most recent call last):
  File "/usr/lib/librephotos/backend/api/directory_watcher.py", line 90, in handle_new_image
    photo._generate_thumbnail()
  File "/usr/lib/librephotos/backend/api/models/photo.py", line 132, in _generate_thumbnail
    with Image(filename=self.image_path) as img:
  File "/usr/lib/librephotos/.local/lib/python3.9/site-packages/wand/image.py", line 9110, in __init__
    self.read(filename=filename)
  File "/usr/lib/librephotos/.local/lib/python3.9/site-packages/wand/image.py", line 9705, in read
    self.raise_exception()
  File "/usr/lib/librephotos/.local/lib/python3.9/site-packages/wand/resource.py", line 222, in raise_exception
    raise e
wand.exceptions.DelegateError: delegate failed `'ffmpeg' -nostdin -v -1 -i '%i' -vframes %S -vcodec pam -an -f rawvideo -y '%u.pam' 2> '%u'' @ error/delegate.c/InvokeDelegate/1966
~~~
After install ffmpeg
~~~
20200118_195052.mp4

reason: cache resources exhausted `/tmp/magick-RKvHmvmInW9R3NWGXP6FveysIVHoWR56.pam' @ error/cache.c/OpenPixelCache/4095
Traceback (most recent call last):
  File "/usr/lib/librephotos/backend/api/directory_watcher.py", line 134, in rescan_image
    photo._generate_thumbnail()
  File "/usr/lib/librephotos/backend/api/models/photo.py", line 132, in _generate_thumbnail
    with Image(filename=self.image_path) as img:
  File "/usr/lib/librephotos/.local/lib/python3.9/site-packages/wand/image.py", line 9110, in __init__
    self.read(filename=filename)
  File "/usr/lib/librephotos/.local/lib/python3.9/site-packages/wand/image.py", line 9705, in read
    self.raise_exception()
  File "/usr/lib/librephotos/.local/lib/python3.9/site-packages/wand/resource.py", line 222, in raise_exception
    raise e
wand.exceptions.CacheError: cache resources exhausted `/tmp/magick-RKvHmvmInW9R3NWGXP6FveysIVHoWR56.pam' @ error/cache.c/OpenPixelCache/4095
~~~

**Comment by tomamplius** [comment]
With ffmpeg big temporary file are store in /tmp/ => no space

**Comment by derneuere** [comment]
Hmm, we could add is_valid_media again and just open the image with wand. If it throws an error return false, if it works return true. What do you think?

**Comment by tomamplius** [comment]
I understand that wand can open mp4 without error if ffmpeg is install. 
Why Mime type start with image not working?

**Comment by derneuere** [comment]
I think this is related to a memory setting see https://github.com/ImageMagick/ImageMagick/issues/396
It tries to load the whole file in memory and apparently the default is not that big.

**Comment by tomamplius** [comment]
imghdr is native way to detect image. It can be use to filter file to process?

~~~
import imghdr
if imghdr.what('path_of_file) :
   print("is image")
else:
   print('not image')
~~~

**Comment by derneuere** [comment]
I don't think that it will detected raw files like .arw or .dng.
It should work properly right now, but will log the file it could not handle.

**Comment by tomamplius** [comment]
you are right!!! i understand the difficulty now.

Mime type is not  "image/dng" or image/arw or image/x-sony-arw like "image/*" ?

Just for information on the case when ffmpeg is available (normaly not available) 
~~~
>>> from wand.image import Image
>>> image_path="/var/lib/librephotos/data/photos/2020/01-janvier/18 janvier les enfoirés avec cynthia/20200118_230616.mp4"
>>> Image(filename=image_path)
.....
long minutes before exception raise
.....
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/librephotos/.local/lib/python3.9/site-packages/wand/image.py", line 9110, in __init__
    self.read(filename=filename)
  File "/usr/lib/librephotos/.local/lib/python3.9/site-packages/wand/image.py", line 9705, in read
    self.raise_exception()
  File "/usr/lib/librephotos/.local/lib/python3.9/site-packages/wand/resource.py", line 222, in raise_exception
    raise e
wand.exceptions.CacheError: cache resources exhausted `/tmp/magick-hz0g7RxusNc1qTk4Gosvg8gmoEWAg0NJ.pam' @ error/cache.c/OpenPixelCache/4095
~~~

Big file are write on /tmp/magick-hz0g7RxusNc1qTk4Gosvg8gmoEWAg0NJ.pam

**Comment by derneuere** [comment]
Maybe we should catch video files and use ffmpeg directly to create thumbnails for videos? https://github.com/kkroening/ffmpeg-python

**Comment by tomamplius** [comment]
ok i will do that

**Comment by darrepac** [comment]
In case it is related, I have the following with docker setup:

`2021-03-30 13:59:44,589 : directory_watcher.py : is_valid_media : 24 : INFO : Could not handle /data/2011/2011-07-09 Piscine de Pythagore/GOPR0063.MP4, because delegate failed `'ffmpeg' -nostdin -v -1 -i '%i' -vframes %S -vcodec pam -an -f rawvideo -y '%u.pam' 2> '%u'' @ error/delegate.c/InvokeDelegate/1928`

Is ffmpeg well installed in the docker image?

**Comment by tomamplius** [comment]
currently no. if we add, video will be process for nothing.. We need to implement video media

**Comment by thequailman** [comment]
Does this mean LibrePhotos doesn't index videos currently?  Would explain why I'm missing videos from my library.

I am hitting the same error: 

```
Could not handle /var/lib/librephotos/data/2016/160107-VID_20160107_185851.mp4, because cache resources exhausted `/var/lib/librephotos/tmp/magick-12915RUhKhW32jQK8.pam' @ error/cache.c/OpenPixelCache/4083`
```

I changed the tmp from `/tmp` to `/var/lib/librephotos/tmp` to avoid running out of space (`/tmp` is tmpfs), and that didn't seem to fix it.  (I used `MAGICK_TEMPORARY_PATH=/var/lib/librephotos/tmp` environment variable)
